libopae/plugins/xfpga/sysfs.c

```diff
@@ -301,7 +306,17 @@ STATIC int sysfs_region_destroy(sysfs_fpga_region *region)
 
 int sysfs_region_count(void)
 {
-	return _sysfs_region_count;
+	int res = 0;
+	if (pthread_mutex_lock(&_sysfs_region_lock)) {
```

Comment:
Have a look at opae_mutex_lock(res, mtxptr) to see how you like it.  It will save you some time.